### DISCUSSION TITLE
Interface Templates on Module Type

### DISCUSSION THREAD

**Discussion by chaositytriumph** [netbox-community/netbox#13811] [Help Wanted!]
I'm looking for some help with getting the interface template syntax to work for the Lantronix SLC80002201S. This console server's model number changes based upon how many port modules are installed. 

There are three module bays (A,B,C) and each port module has 16 ports on it. Is there a way so that if any of the modules go into module bay A the ports are 1-16, B = 17-32, and C = 33-48? I'm not sure if I need to utilize the "position" configuration or not.

I've been able to utilize some of the functions with interface templates, but I'm unsure of syntax and configuration order to achieve this.

Thanks!

**Comment by candlerb** [comment]
> This console server's model number changes based upon how many port modules are installed.

In that case it might be simplest to model it as three different device types, with no module bays.

There's a minor issue if you're able to upgrade it to a different model by buying a new module: you can change the device type of an existing device, but it won't automatically inherit new ports. This can be solved by running [a script](https://github.com/netbox-community/customizations/blob/master/scripts/add_device_type_components.py).

> Is there a way so that if any of the modules go into module bay A the ports are 1-16, B = 17-32, and C = 33-48?

Not that I know of. An alternative solution would be to have three different module types, and you pick the right one according to the module bay it goes into.

**Reply by chaositytriumph** [reply]
Thank you. I'll play around with that script a bit later and let you know how that works.  I had that thought myself, but was trying to avoid it if possible.
